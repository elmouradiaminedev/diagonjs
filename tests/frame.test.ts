import { expect, describe, it } from "vitest";
import Diagon, { FrameTranslationOptions } from "src";

function generateAllOptions(): FrameTranslationOptions[] {
  const allOptions: FrameTranslationOptions[] = [];

  for (const asciiOnly of [true, false]) {
    for (const lineNumber of [true, false]) {
      allOptions.push({
        asciiOnly,
        lineNumber,
      });
    }
  }

  return allOptions;
}

describe("Frame expression translation", () => {
  generateAllOptions().forEach((options) => {
    describe(`With options ${JSON.stringify(options)}`, () => {
      it("should translate code frame hello world", () => {
        expect(
          Diagon.translate.frame(
            '#include <iostream>\nusing namespace std;\n\nint main() \n{\n    cout << "Hello, World!";\n    return 0;\n}',
            options,
          ),
        ).toMatchSnapshot();
      });
    });
  });
});
