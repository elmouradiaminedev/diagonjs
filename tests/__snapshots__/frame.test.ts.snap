// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Frame expression translation > With options {"asciiOnly":false,"lineNumber":false} > should translate code frame hello world 1`] = `
"┌────────────────────────────┐
│#include <iostream>         │
│using namespace std;        │
│                            │
│int main()                  │
│{                           │
│    cout << "Hello, World!";│
│    return 0;               │
│}                           │
└────────────────────────────┘
"
`;

exports[`Frame expression translation > With options {"asciiOnly":false,"lineNumber":true} > should translate code frame hello world 1`] = `
"┌─┬────────────────────────────┐
│1│#include <iostream>         │
│2│using namespace std;        │
│3│                            │
│4│int main()                  │
│5│{                           │
│6│    cout << "Hello, World!";│
│7│    return 0;               │
│8│}                           │
└─┴────────────────────────────┘
"
`;

exports[`Frame expression translation > With options {"asciiOnly":true,"lineNumber":false} > should translate code frame hello world 1`] = `
" ____________________________ 
│                            │
│#include <iostream>         │
│using namespace std;        │
│                            │
│int main()                  │
│{                           │
│    cout << "Hello, World!";│
│    return 0;               │
│}                           │
│____________________________│
"
`;

exports[`Frame expression translation > With options {"asciiOnly":true,"lineNumber":true} > should translate code frame hello world 1`] = `
" ______________________________ 
│ |                            │
│1|#include <iostream>         │
│2|using namespace std;        │
│3|                            │
│4|int main()                  │
│5|{                           │
│6|    cout << "Hello, World!";│
│7|    return 0;               │
│8|}                           │
│_|____________________________│
"
`;
